src/main/java/bio/terra/janitor/service/cleanup/flight/GoogleBlobCleanupStep.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.janitor.service.cleanup.flight;
+
+import bio.terra.cloudres.common.ClientConfig;
+import bio.terra.cloudres.google.storage.BlobCow;
+import bio.terra.generated.model.CloudResourceUid;
+import bio.terra.janitor.db.JanitorDao;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.StorageException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Step to cleanup Google Blob resource. */
+public class GoogleBlobCleanupStep extends ResourceCleanupStep {
+  private Logger logger = LoggerFactory.getLogger(GoogleBlobCleanupStep.class);
+
+  public GoogleBlobCleanupStep(ClientConfig clientConfig, JanitorDao janitorDao) {
+    super(clientConfig, janitorDao);
+  }
+
+  @Override
+  protected StepResult cleanUp(CloudResourceUid resourceUid) {
+    try {
+      // Delete all Blobs before deleting bucket.
+      BlobId blobId =
+          BlobId.of(
+              resourceUid.getGoogleBlobUid().getBucketName(),
+              resourceUid.getGoogleBlobUid().getBlobName());
+      BlobCow blobCow = storageCow.get(blobId);
+      if (blobCow == null) {
+        // Blob doesn't exists.
+        return StepResult.getStepResultSuccess();
+      }
+      blobCow.delete();
```

Comment:
what happens if we skip the `get` and just try to delete the blob directly?

Reply:
Ah, right, remove the extra get